### DISCUSSION TITLE
PyAssimp - How to interpret the material values?

### DISCUSSION THREAD

**Discussion by sourcehawk** [assimp/assimp#4192] [Q&A] [answered]
I'm wondering how I am supposed to interpret the data I receive in the "materials" array?

I'm loading a Wavefront (.obj + .mtl) file and the diffuse and specular properties which I expected to be a set of 3 floats are all single integers. I saw a code example somewhere in this repository ([here](https://github.com/assimp/assimp/blob/master/port/PyAssimp/scripts/sample.py)) which seemed to suggest that this was the expected output (a single integer) but I am no closer to understanding what to do with it.

Snippet from the .mtl file:
```
newmtl Gold
Ka 0.000000 0.000000 0.000000
Kd 0.803922 0.698039 0.094118
Ks 0.330000 0.330000 0.330000
```
Snippet of the material values from the debugger
![image](https://user-images.githubusercontent.com/54936225/142717134-2b66a558-922b-4929-9ec2-59aa82a687e5.png)

**Comment by kimkulling** [comment]
You can just set the value ka for the ambient color of the material. Kd stands for the diffuse color and Ks for the specular color.

**Reply by sourcehawk** [reply]
You are saying colors, but a color as far as I know is represented with RGB values which consists of 3 numbers each. I don't understand why the values are transformed to integers or how I should use these integers to calculate colors in the shader.

I'm implementing my own shaders in OpenGL where ambient, diffuse and specular need to be vectors with 3 floats, just as they are presented in the MTL file in order to perform fragment calculations.

**Comment by kimkulling** [comment, marked as answer]
All data in a material is stored as a parameter. So if you want to see what kind of data is stored in your material-instance I strongly recommend to study our samples, like:

```
 print("MATERIALS:")
    for index, material in enumerate(scene.materials):
        print("  MATERIAL (id:" + str(index+1) + ")")
        for key, value in material.properties.items():
            print("    %s: %s" % (key, value))
```

So you can reach each data-item like the diffuce color by its key and the value contains the data assigned to this. As far as I know colors are stored as Color3D, which looks like:

```
class Color3D(Structure):
    """
    See 'types.h' for details.
    """

    _fields_ = [
            #  Red, green and blue color values
            ("r", c_float),("g", c_float),("b", c_float),
        ]

```

So if I get the sample right you shall be able to access all the components by using r, g, b
